### DISCUSSION TITLE
Send custom metadata to a live stream

### DISCUSSION THREAD

**Discussion by bypass112** [shogo4405/HaishinKit.dart#24] [Ideas]
Hello, first of all thank you for creating this Flutter package!
I've seen that rtmp support metadata push, alongside the video publishing stream.
I've looked over the [Haishinkit.kt](https://github.com/shogo4405/HaishinKit.kt/blob/d83adf67d567ab6d9bac030ecb5821da7f0fc0a1/haishinkit/src/main/java/com/haishinkit/rtmp/RtmpStream.kt#L204) and [Haishinkit.swift](https://github.com/shogo4405/HaishinKit.swift/blob/main/Sources/RTMP/RTMPStream.swift#L554) and it seems that both can send metadata as per Adobe's documentation, using [ns.send("@setDataFrame", "onMetaData", metaData);](https://helpx.adobe.com/adobe-media-server/dev/adding-metadata-live-stream.html).
Is there an already implemented way to send / update custom metadata to a live stream that I am missing?
If there isn't what are the steps to implement a way to send rtmp metadata through Flutter?
Thank you!

**Comment by shogo4405** [comment]
I haven't made it available for use on the Flutter side. Also, on the native side, it seems that updates cannot be made with the current implementation. Sends it works ok.
